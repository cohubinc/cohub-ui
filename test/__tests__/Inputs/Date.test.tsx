import React from "react";
import { Inputs } from "dist";

import "@testing-library/jest-dom/extend-expect";
import { mount } from "enzyme";

describe("Date", () => {
  it("matches snapshot", async () => {
    const wrapper = mount(
      <Inputs.Date
        label="Date"
        appearance="contrast"
        input={{
          name: "nunyah",
          value: "1984-09-09",
          onChange: () => null,
          onFocus: () => null,
          onBlur: () => null
        }}
        meta={{}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  // Don't think we need these test b/c they're pretty much duplicates from the DatePickerBase test, but they're here if that proves to be wrong //
  /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

  //   it("Shows a properly formated date on init when at ISO 8601 formated datetime value is passed in", () => {
  //   const { getByLabelText } = render(
  //     <Date
  //       label="Date"
  //       appearance="contrast"
  //       input={{ value: "1984-09-09", onChange: () => null }}
  //     />
  //   );
  //   const input = getByLabelText("Date") as HTMLInputElement;

  //   expect(input.value).toBe("09-09-1984");
  // });

  // it("It works when empty value is passed in", () => {
  //   const { getByLabelText } = render(
  //     <Date
  //       label="Date"
  //       appearance="contrast"
  //       input={{ value: "", onChange: () => null }}
  //     />
  //   );
  //   const input = getByLabelText("Date") as HTMLInputElement;

  //   expect(input.value).toBe("");
  // });

  // it("Typing a valid date into input works", async () => {
  //   const onChangeFn = jest.fn(isoDateTime => isoDateTime);
  //   const { getByLabelText } = render(
  //     <Date
  //       label="Date"
  //       appearance="contrast"
  //       input={{ value: "", onChange: onChangeFn }}
  //     />
  //   );
  //   const input = getByLabelText("Date") as HTMLInputElement;

  //   fireEvent.change(input, { target: { value: "09091999" } });

  //   expect(input.value).toBe("09-09-1999");

  //   await sleep();

  //   expect(onChangeFn.mock.calls.length).toBe(1);
  //   expect(onChangeFn.mock.calls[0][0]).toBe("1999-09-09");
  // });

  // it("Typing non-numeric chars doesn't get you very far", () => {
  //   const { getByLabelText } = render(
  //     <Date
  //       label="Date"
  //       appearance="contrast"
  //       input={{ value: "", onChange: () => null }}
  //     />
  //   );
  //   const input = getByLabelText("Date") as HTMLInputElement;

  //   fireEvent.change(input, { target: { value: "ABCD@#$%{}09" } });

  //   expect(input.value).toBe("09-DD-YYYY");
  // });

  // it("Typing an out of range year doesn't call the onChange callback", async () => {
  //   const onChangeFn = jest.fn(isoDateTime => isoDateTime);
  //   const { getByLabelText } = render(
  //     <Date
  //       label="Date"
  //       appearance="contrast"
  //       input={{ value: "", onChange: onChangeFn }}
  //     />
  //   );
  //   const input = getByLabelText("Date") as HTMLInputElement;

  //   fireEvent.change(input, { target: { value: "09091900" } });

  //   await sleep();

  //   expect(onChangeFn.mock.calls.length).toBe(0);
  // });

  // it("Typing an out of range date triggers the error state", async () => {
  //   const onChangeFn = jest.fn(isoDateTime => isoDateTime);
  //   const { getByLabelText } = render(
  //     <Date
  //       label="Date"
  //       appearance="contrast"
  //       input={{ value: "", onChange: onChangeFn }}
  //     />
  //   );
  //   const input = getByLabelText("Date") as HTMLInputElement;

  //   fireEvent.change(input, { target: { value: "22011999" } });

  //   expect(input).toBeInvalid();
  // });

  // it("Picker isn't visible on init", async () => {
  //   const { getByTestId } = render(
  //     <Date
  //       label="Date"
  //       appearance="contrast"
  //       input={{ value: "1984-09-09", onChange: () => null }}
  //     />
  //   );

  //   expect(getByTestId("pickerContainer").getAttribute("aria-hidden")).toBe(
  //     "true"
  //   );
  // });

  // it("Picker is visible after input is focused", async () => {
  //   const { getByTestId, getByLabelText } = render(
  //     <Date
  //       label="Date"
  //       appearance="contrast"
  //       input={{ value: "1984-09-09", onChange: () => null }}
  //     />
  //   );

  //   const input = getByLabelText("Date") as HTMLInputElement;

  //   fireEvent.focus(input);

  //   await sleep();

  //   expect(getByTestId("pickerContainer").getAttribute("aria-hidden")).toBe(
  //     "false"
  //   );
  // });

  // it("Selecting a month from the month picker updates the input value correctly", async () => {
  //   const { getByLabelText, findBtnByText } = render(
  //     <Date
  //       label="Date"
  //       appearance="contrast"
  //       input={{ value: "", onChange: () => null }}
  //     />
  //   );

  //   const input = getByLabelText("Date") as HTMLInputElement;

  //   fireEvent.focus(input);

  //   const mayBtn = await findBtnByText("May");

  //   mayBtn.click();

  //   expect(input.value).toBe("05-DD-YYYY");
  // });

  // it("Selecting a month and then a day from the month picker updates the input value correctly", async () => {
  //   const { getByLabelText, findBtnByText } = render(
  //     <Date
  //       label="Date"
  //       appearance="contrast"
  //       input={{ value: "", onChange: () => null }}
  //     />
  //   );

  //   const input = getByLabelText("Date") as HTMLInputElement;

  //   fireEvent.focus(input);

  //   const mayBtn = await findBtnByText("May");

  //   mayBtn.click();

  //   const daySevenBtn = await findBtnByText("07");

  //   daySevenBtn.click();

  //   expect(input.value).toBe("05-07-YYYY");
  // });

  // it("Selecting a full date from the picker UI updates the input value correctly and call the onChange callback", async () => {
  //   const onChangeFn = jest.fn(isoDateTime => isoDateTime);
  //   const { getByLabelText, findBtnByText } = render(
  //     <Date
  //       label="Date"
  //       appearance="contrast"
  //       input={{ value: "", onChange: onChangeFn }}
  //     />
  //   );

  //   const input = getByLabelText("Date") as HTMLInputElement;

  //   fireEvent.focus(input);

  //   const mayBtn = await findBtnByText("May");

  //   mayBtn.click();

  //   const daySevenBtn = await findBtnByText("07");

  //   daySevenBtn.click();

  //   const yearBtn = await findBtnByText("2019");

  //   yearBtn.click();

  //   await sleep();

  //   expect(onChangeFn.mock.calls.length).toBe(1);
  //   expect(onChangeFn.mock.calls[0][0]).toBe("2019-05-07");
  // });
});
