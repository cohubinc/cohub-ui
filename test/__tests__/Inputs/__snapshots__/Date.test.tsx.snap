// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date matches snapshot 1`] = `
<Date$1
  appearance="contrast"
  input={
    Object {
      "name": "nunyah",
      "onBlur": [Function],
      "onChange": [Function],
      "onFocus": [Function],
      "value": "1984-09-09",
    }
  }
  label="Date"
  meta={Object {}}
>
  <DatePickerBase
    appearance="contrast"
    label="Date"
    name="nunyah"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    value="1984-09-09"
  >
    <FloatingLabelWrapper
      appearance="contrast"
      floatLabel={true}
      label="Date"
      name="nunyah"
      onBlur={[Function]}
      onFocus={[Function]}
      value="1984-09-09"
    >
      <div
        className="FloatingLabelWrapper ContrastInput "
        style={
          Object {
            "color": "hsl(0, 2%, 12%)",
            "cursor": "text",
            "position": "relative",
          }
        }
      >
        <div
          className="inputWrapper  bd-radius"
        >
          <LowLevelDatePicker
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            rangeError={false}
            setHasValue={[Function]}
            setNativeElRef={[Function]}
            setRangeError={[Function]}
            style={
              Object {
                "color": "hsl(0, 2%, 12%)",
                "cursor": "text",
                "textAlign": undefined,
              }
            }
            value="1984-09-09"
          >
            <div
              className=""
              style={
                Object {
                  "color": "hsl(0, 2%, 12%)",
                  "cursor": "text",
                  "position": "relative",
                  "textAlign": undefined,
                }
              }
            >
              <NumberFormat
                allowEmptyFormatting={false}
                allowLeadingZeros={false}
                allowNegative={true}
                aria-invalid={false}
                className="LowLevelDatePickerInput LowLevelDatePicker-module_input__JwcFZ"
                data-testid="LowLevelDatePickerInput"
                decimalSeparator="."
                displayType="input"
                fixedDecimalScale={false}
                format="##-##-####"
                getInputRef={[Function]}
                isAllowed={[Function]}
                isNumericString={false}
                mask={
                  Array [
                    "M",
                    "M",
                    "D",
                    "D",
                    "Y",
                    "Y",
                    "Y",
                    "Y",
                  ]
                }
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseUp={[Function]}
                onValueChange={[Function]}
                prefix=""
                suffix=""
                thousandsGroupStyle="thousand"
                type="text"
                value="09-09-1984"
              >
                <input
                  aria-invalid={false}
                  className="LowLevelDatePickerInput LowLevelDatePicker-module_input__JwcFZ"
                  data-testid="LowLevelDatePickerInput"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseUp={[Function]}
                  type="text"
                  value="09-09-1984"
                />
              </NumberFormat>
              <Picker
                dateTime={
                  Object {
                    "day": "09",
                    "month": "09",
                    "year": "1984",
                  }
                }
                goToSection={[Function]}
                minDate={
                  Object {
                    "minDay": 1,
                    "minMonth": 1,
                    "minYear": 1980,
                  }
                }
                open={false}
                picker="month"
                setMonth={[Function]}
                setMonthAndDay={[Function]}
                setYear={[Function]}
                style={
                  Object {
                    "position": "absolute",
                    "top": 50,
                  }
                }
              >
                <AnimateHeight
                  animateOpacity={false}
                  animationStateClasses={
                    Object {
                      "animating": "rah-animating",
                      "animatingDown": "rah-animating--down",
                      "animatingToHeightAuto": "rah-animating--to-height-auto",
                      "animatingToHeightSpecific": "rah-animating--to-height-specific",
                      "animatingToHeightZero": "rah-animating--to-height-zero",
                      "animatingUp": "rah-animating--up",
                      "static": "rah-static",
                      "staticHeightAuto": "rah-static--height-auto",
                      "staticHeightSpecific": "rah-static--height-specific",
                      "staticHeightZero": "rah-static--height-zero",
                    }
                  }
                  applyInlineTransitions={true}
                  data-testid="pickerContainer"
                  delay={0}
                  duration={250}
                  easing="ease-in"
                  height={0}
                  style={
                    Object {
                      "position": "absolute",
                      "top": 50,
                      "zIndex": 2,
                    }
                  }
                >
                  <div
                    aria-hidden={true}
                    className="rah-static rah-static--height-zero"
                    data-testid="pickerContainer"
                    style={
                      Object {
                        "height": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "top": 50,
                        "zIndex": 2,
                      }
                    }
                  >
                    <div
                      style={Object {}}
                    >
                      <Segment
                        bordered={false}
                        className=""
                        elevation={1}
                        padded={true}
                        style={
                          Object {
                            "display": "inline-block",
                            "position": "relative",
                          }
                        }
                      >
                        <div
                          className="Segment-module_CohubSegment__3MMwJ Segment-module_padded__39Fvk "
                          style={
                            Object {
                              "backgroundColor": "hsl(0, 0%, 100%)",
                              "border": "",
                              "boxShadow": "0px -1px 1px rgba(0, 0, 0, 0.03), 0px 1px 1px rgba(0, 0, 0, 0.14)",
                              "display": "inline-block",
                              "position": "relative",
                            }
                          }
                        >
                          <div
                            className="flex justify-center align-items-center mb-1"
                          >
                            <DateHeader
                              day="09"
                              month="09"
                              onClick={[Function]}
                              picker="month"
                              year="1984"
                            >
                              <div
                                className="flex"
                              >
                                <MonthUnit
                                  active={true}
                                  onClick={[Function]}
                                >
                                  <AnimateHeight
                                    animateOpacity={false}
                                    animationStateClasses={
                                      Object {
                                        "animating": "rah-animating",
                                        "animatingDown": "rah-animating--down",
                                        "animatingToHeightAuto": "rah-animating--to-height-auto",
                                        "animatingToHeightSpecific": "rah-animating--to-height-specific",
                                        "animatingToHeightZero": "rah-animating--to-height-zero",
                                        "animatingUp": "rah-animating--up",
                                        "static": "rah-static",
                                        "staticHeightAuto": "rah-static--height-auto",
                                        "staticHeightSpecific": "rah-static--height-specific",
                                        "staticHeightZero": "rah-static--height-zero",
                                      }
                                    }
                                    applyInlineTransitions={true}
                                    delay={0}
                                    duration={250}
                                    easing="ease"
                                    height="auto"
                                    style={
                                      Object {
                                        "display": "inline-block",
                                      }
                                    }
                                  >
                                    <div
                                      aria-hidden={false}
                                      className="rah-static rah-static--height-auto"
                                      style={
                                        Object {
                                          "display": "inline-block",
                                          "height": "auto",
                                          "overflow": "visible",
                                        }
                                      }
                                    >
                                      <div
                                        style={Object {}}
                                      >
                                        <DateUnitButton
                                          active={true}
                                          onClick={[Function]}
                                        >
                                          <Blank
                                            className="SectionNavigation-module_action__wr6E9 SectionNavigation-module_active__3scI6"
                                            onClick={[Function]}
                                            tabIndex={-1}
                                          >
                                            <button
                                              className="Blank-module_ButtonBlank__2u_7K SectionNavigation-module_action__wr6E9 SectionNavigation-module_active__3scI6"
                                              onClick={[Function]}
                                              tabIndex={-1}
                                            >
                                              <Typography
                                                style={Object {}}
                                              >
                                                <Component
                                                  style={Object {}}
                                                >
                                                  <span
                                                    className=""
                                                    data-qa="text"
                                                    style={
                                                      Object {
                                                        "color": "hsl(0, 2%, 22%)",
                                                        "fontFamily": "Inter",
                                                        "fontSize": "14px",
                                                        "fontWeight": 400,
                                                        "textAlign": "initial",
                                                        "textTransform": "initial",
                                                      }
                                                    }
                                                  >
                                                    <Typography
                                                      bold={true}
                                                    >
                                                      <Component
                                                        bold={true}
                                                      >
                                                        <span
                                                          className=""
                                                          data-qa="text"
                                                          style={
                                                            Object {
                                                              "color": "hsl(0, 2%, 22%)",
                                                              "fontFamily": "Inter",
                                                              "fontSize": "14px",
                                                              "fontWeight": 600,
                                                              "textAlign": "initial",
                                                              "textTransform": "initial",
                                                            }
                                                          }
                                                        >
                                                          September
                                                        </span>
                                                      </Component>
                                                    </Typography>
                                                  </span>
                                                </Component>
                                              </Typography>
                                            </button>
                                          </Blank>
                                        </DateUnitButton>
                                         
                                      </div>
                                    </div>
                                  </AnimateHeight>
                                </MonthUnit>
                                <DayUnit
                                  active={false}
                                  onClick={[Function]}
                                >
                                  <AnimateHeight
                                    animateOpacity={false}
                                    animationStateClasses={
                                      Object {
                                        "animating": "rah-animating",
                                        "animatingDown": "rah-animating--down",
                                        "animatingToHeightAuto": "rah-animating--to-height-auto",
                                        "animatingToHeightSpecific": "rah-animating--to-height-specific",
                                        "animatingToHeightZero": "rah-animating--to-height-zero",
                                        "animatingUp": "rah-animating--up",
                                        "static": "rah-static",
                                        "staticHeightAuto": "rah-static--height-auto",
                                        "staticHeightSpecific": "rah-static--height-specific",
                                        "staticHeightZero": "rah-static--height-zero",
                                      }
                                    }
                                    applyInlineTransitions={true}
                                    delay={0}
                                    duration={250}
                                    easing="ease"
                                    height="auto"
                                    style={
                                      Object {
                                        "display": "inline-block",
                                      }
                                    }
                                  >
                                    <div
                                      aria-hidden={false}
                                      className="rah-static rah-static--height-auto"
                                      style={
                                        Object {
                                          "display": "inline-block",
                                          "height": "auto",
                                          "overflow": "visible",
                                        }
                                      }
                                    >
                                      <div
                                        style={Object {}}
                                      >
                                        <DateUnitButton
                                          active={false}
                                          onClick={[Function]}
                                        >
                                          <Blank
                                            className="SectionNavigation-module_action__wr6E9 "
                                            onClick={[Function]}
                                            tabIndex={-1}
                                          >
                                            <button
                                              className="Blank-module_ButtonBlank__2u_7K SectionNavigation-module_action__wr6E9 "
                                              onClick={[Function]}
                                              tabIndex={-1}
                                            >
                                              <Typography
                                                style={Object {}}
                                              >
                                                <Component
                                                  style={Object {}}
                                                >
                                                  <span
                                                    className=""
                                                    data-qa="text"
                                                    style={
                                                      Object {
                                                        "color": "hsl(0, 2%, 22%)",
                                                        "fontFamily": "Inter",
                                                        "fontSize": "14px",
                                                        "fontWeight": 400,
                                                        "textAlign": "initial",
                                                        "textTransform": "initial",
                                                      }
                                                    }
                                                  >
                                                    <Typography
                                                      bold={true}
                                                    >
                                                      <Component
                                                        bold={true}
                                                      >
                                                        <span
                                                          className=""
                                                          data-qa="text"
                                                          style={
                                                            Object {
                                                              "color": "hsl(0, 2%, 22%)",
                                                              "fontFamily": "Inter",
                                                              "fontSize": "14px",
                                                              "fontWeight": 600,
                                                              "textAlign": "initial",
                                                              "textTransform": "initial",
                                                            }
                                                          }
                                                        >
                                                          09
                                                        </span>
                                                      </Component>
                                                    </Typography>
                                                  </span>
                                                </Component>
                                              </Typography>
                                            </button>
                                          </Blank>
                                        </DateUnitButton>
                                      </div>
                                    </div>
                                  </AnimateHeight>
                                </DayUnit>
                                <YearUnit
                                  active={false}
                                  onClick={[Function]}
                                >
                                  <AnimateHeight
                                    animateOpacity={false}
                                    animationStateClasses={
                                      Object {
                                        "animating": "rah-animating",
                                        "animatingDown": "rah-animating--down",
                                        "animatingToHeightAuto": "rah-animating--to-height-auto",
                                        "animatingToHeightSpecific": "rah-animating--to-height-specific",
                                        "animatingToHeightZero": "rah-animating--to-height-zero",
                                        "animatingUp": "rah-animating--up",
                                        "static": "rah-static",
                                        "staticHeightAuto": "rah-static--height-auto",
                                        "staticHeightSpecific": "rah-static--height-specific",
                                        "staticHeightZero": "rah-static--height-zero",
                                      }
                                    }
                                    applyInlineTransitions={true}
                                    delay={0}
                                    duration={250}
                                    easing="ease"
                                    height="auto"
                                    style={
                                      Object {
                                        "display": "inline-block",
                                      }
                                    }
                                  >
                                    <div
                                      aria-hidden={false}
                                      className="rah-static rah-static--height-auto"
                                      style={
                                        Object {
                                          "display": "inline-block",
                                          "height": "auto",
                                          "overflow": "visible",
                                        }
                                      }
                                    >
                                      <div
                                        style={Object {}}
                                      >
                                        ,
                                         
                                        <DateUnitButton
                                          active={false}
                                          onClick={[Function]}
                                        >
                                          <Blank
                                            className="SectionNavigation-module_action__wr6E9 "
                                            onClick={[Function]}
                                            tabIndex={-1}
                                          >
                                            <button
                                              className="Blank-module_ButtonBlank__2u_7K SectionNavigation-module_action__wr6E9 "
                                              onClick={[Function]}
                                              tabIndex={-1}
                                            >
                                              <Typography
                                                style={Object {}}
                                              >
                                                <Component
                                                  style={Object {}}
                                                >
                                                  <span
                                                    className=""
                                                    data-qa="text"
                                                    style={
                                                      Object {
                                                        "color": "hsl(0, 2%, 22%)",
                                                        "fontFamily": "Inter",
                                                        "fontSize": "14px",
                                                        "fontWeight": 400,
                                                        "textAlign": "initial",
                                                        "textTransform": "initial",
                                                      }
                                                    }
                                                  >
                                                    <Typography
                                                      bold={true}
                                                    >
                                                      <Component
                                                        bold={true}
                                                      >
                                                        <span
                                                          className=""
                                                          data-qa="text"
                                                          style={
                                                            Object {
                                                              "color": "hsl(0, 2%, 22%)",
                                                              "fontFamily": "Inter",
                                                              "fontSize": "14px",
                                                              "fontWeight": 600,
                                                              "textAlign": "initial",
                                                              "textTransform": "initial",
                                                            }
                                                          }
                                                        >
                                                          1984
                                                        </span>
                                                      </Component>
                                                    </Typography>
                                                  </span>
                                                </Component>
                                              </Typography>
                                            </button>
                                          </Blank>
                                        </DateUnitButton>
                                      </div>
                                    </div>
                                  </AnimateHeight>
                                </YearUnit>
                              </div>
                            </DateHeader>
                          </div>
                          <div
                            style={
                              Object {
                                "height": 300,
                                "width": 300,
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "display": "flex",
                                  "flex": 1,
                                  "overflow": "hidden",
                                }
                              }
                            >
                              <div
                                id="section-row"
                                style={
                                  Object {
                                    "display": "flex",
                                    "minHeight": "min-content",
                                  }
                                }
                              >
                                <Section
                                  open={false}
                                  style={
                                    Object {
                                      "marginLeft": -0,
                                      "transition": "all 350ms ease-in",
                                    }
                                  }
                                />
                                <Section
                                  open={false}
                                />
                                <Section
                                  open={false}
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </Segment>
                    </div>
                  </div>
                </AnimateHeight>
              </Picker>
            </div>
          </LowLevelDatePicker>
          <span
            className="bar "
          />
        </div>
        <label
          className="FloatedLabel"
          data-qa="base-input-element-label"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": undefined,
              "color": "hsl(356, 3%, 59%)",
              "cursor": "text",
              "width": undefined,
            }
          }
        >
          Date
        </label>
      </div>
    </FloatingLabelWrapper>
  </DatePickerBase>
</Date$1>
`;
